buildtools/DevConfigGenerator/DevConfigGenerator/Program.cs

```diff
@@ -0,0 +1,209 @@
+            }
+
+            AnsiConsole.Write(table);
+        }
+
+        var changeType = AnsiConsole.Prompt(
+            new SelectionPrompt<string>()
+                .Title("What type of change was made?")
+                .AddChoices(new[] {
+                "Core", "Generator", "Service", "Save DevConfigFile", "Exit Without Saving"
+                }));
+
+        switch (changeType)
+        {
+            case "Core":
+                CoreChange();
+                break;
+            case "Generator":
+                GeneratorChange();
+                break;
+            case "Service":
+                ServiceChange();
+                break;
+            case "Save DevConfigFile":
+                WriteDevConfigFile();
+                return;
+            case "Exit Without Saving":
+                return;
+        }
+    }
+
+}
+catch (Exception e)
+{
+    Console.WriteLine("Unhandled exception while generating dev config file");
+    Console.WriteLine(e.Message);
+}
+
+void WriteDevConfigFile()
+{
+    if (devConfig.Core == null && !devConfig.Services.Any())
+    {
+        AnsiConsole.Markup("[yellow]Skipped writing dev config since the config is empty[/]\n");
+    }
+
+    var devConfigDirectory = Path.Combine(FindRepoRoot(), "generator", ".DevConfigs");
+    if (!Directory.Exists(devConfigDirectory))
+    {
+        Directory.CreateDirectory(devConfigDirectory);
+    }
+
+    var devConfigFilePath = Path.Combine(devConfigDirectory, Guid.NewGuid().ToString() + ".json");
+    DevConfigWriter.Write(devConfig, devConfigFilePath);
+
+    AnsiConsole.Markup($"DevConfig file written to: [green]{devConfigFilePath}[/]\n");
+    AnsiConsole.WriteLine(File.ReadAllText(devConfigFilePath));
+}
+
+
+void CoreChange()
+{
+    var change = AnsiConsole.Prompt(new TextPrompt<string>("Enter the change message for the [green]core[/] change:"));
+
+    var newCoreConfig = false;
+    if (devConfig.Core == null)
+    {
+        newCoreConfig = true;
+        devConfig.Core = new CoreChange();
+    }
+
+    devConfig.Core.Messages.Add(change);
+
+    devConfig.Core.Type = AskChangeType(devConfig.Core.Type);
+
+    var textPrompt = new TextPrompt<bool>("Should service packages be updated to use new Core?")
+            .AddChoice(true)
+            .AddChoice(false)
+            .WithConverter(choice => choice ? "y" : "n");
+
+    // If added a second change message to core then default the update service minimum to the value choosen for the previous entry.
+    if (!newCoreConfig)
+    {
+        textPrompt.DefaultValue(devConfig.Core.UpdateServiceMinimum);
+    }
+
+    var updateServiceMinimum = AnsiConsole.Prompt(textPrompt);
+
+    devConfig.Core.UpdateServiceMinimum = updateServiceMinimum;
+}
+
+void GeneratorChange()
+{
+    AnsiConsole.Markup("Generator changes are saved as [green]Core changes[/] with update service minimum set to [green]true[/].\n");
```

Comment:
A generator change doesn't always means a core update is needed. Usually it requires a change to a specific service unless it impacts many of them.

Reply:
Agreed but in those cases a user doesn't have to choose the generator option and could just choose the affect service. I had this option because it felt like the safe way to handle generator changes and avoid people making generator changes that did affect services they didn't realize and not get released.